Remove support for secp192[k|r]1 curves

valeriosetti opened the pull request:
## Description

Resolves #568 

This PR depends on:
- #513 
- #564 
- #565
- https://github.com/Mbed-TLS/mbedtls-framework/pull/242
- https://github.com/Mbed-TLS/mbedtls/pull/10522

## PR checklist

- [x] **changelog** provided
- [x] **framework PR** not required
- [x] **mbedtls development PR** provided:
    - https://github.com/Mbed-TLS/mbedtls/pull/10519
    - https://github.com/Mbed-TLS/mbedtls/pull/10522
- [x] **mbedtls 3.6 PR** not required because: no backport
- **tests**  not required because: already updated in prerequisite PRs

valeriosetti committed:
include: crypto_config: remove PSA_WANTs for secp192 curves

Signed-off-by: Valerio Setti <vsetti@baylibre.com>

valeriosetti committed:
include|core: remove support for secp192[k|r]1 curves

The most important change here is that now "tf_psa_crypto_check_config.h"
will error out if PSA_WANT_ECC_SECP_K1_192 or
PSA_WANT_ECC_SECP_K1_192 are enabled in the build, whereas previously
they could still be enabled if TF_PSA_CRYPTO_ALLOW_REMOVED_MECHANISMS was
enabled or in case they were supported while building libtestdriver1.

Signed-off-by: Valerio Setti <vsetti@baylibre.com>

valeriosetti committed:
drivers: include: ecp: remove support for secp192 curves

Signed-off-by: Valerio Setti <vsetti@baylibre.com>

valeriosetti committed:
drivers: psa_util: remove support for secp192 curves

Signed-off-by: Valerio Setti <vsetti@baylibre.com>

valeriosetti committed:
drivers: oid: remove support for secp192 curves

Signed-off-by: Valerio Setti <vsetti@baylibre.com>

valeriosetti committed:
drivers: builtin: completely remove any support for secp192 curves

Signed-off-by: Valerio Setti <vsetti@baylibre.com>

mpg commented:
CI's unhappy:
```
/var/lib/build/tf-psa-crypto/core/tf_psa_crypto_check_config.h:123:48: error: expected end of line in preprocessor expression
      defined(MBEDTLS_ECP_DP_BP512R1_ENABLED) )
```

bjwtaylor commented on ChangeLog.d/remove-secp192-curves.txt:
```diff
@@ -0,0 +1,4 @@
+Removals
+   * Support for secp192[r|k]1 curves was previously disabled by deafult and
```
Presumably default and not deafult

mpg replied:
I'm not even sure this needs a ChangeLog entry. The 1.0 ChangeLog already says:
```
   * Support for EC curves secp224r1, secp224k1, secp192k1 and secp192r1 is
     removed.
```
The fact that the curves were still present, but disabled by default, is an undocumented detail so I don't think we need to announce this change.

Note: the curves were not just disabled by default, re-enabling them required to define the also undocumented flag `TF_PSA_CRYPTO_ALLOW_REMOVED_MECHANISMS` so that's not something that could happen by accident.

valeriosetti replied:
Since this was the very last commit (at least for now), I thought it was nicer to just [remove it from the list](https://github.com/Mbed-TLS/TF-PSA-Crypto/pull/570#event-21362870484) and then address other comments instead of adding noise. I hope you won't mind

bjwtaylor commented on core/tf_psa_crypto_check_config.h:
```diff
@@ -596,9 +590,7 @@
-#if (defined(PSA_WANT_ECC_SECP_K1_192) || defined(PSA_WANT_ECC_SECP_K1_192)) \
-    && !defined(TF_PSA_CRYPTO_ALLOW_REMOVED_MECHANISMS) \
-    && !defined(TF_PSA_CRYPTO_WE_ARE_IN_LIBTESTDRIVER1)
+#if (defined(PSA_WANT_ECC_SECP_K1_192) || defined(PSA_WANT_ECC_SECP_K1_192)) //no-check-names
```
Is there any reason this can't just be #if (defined(PSA_WANT_ECC_SECP_K1_192)), as both arguments of the or operator are the same?

mpg replied:
I think one of them was supposed to be R1 not K1 (same in the error message below, pre-existing but let's fix it now that Ben noticed it).

valeriosetti replied:
Right, I totally missed this when I working on this change. Thanks for the heads up!

bjwtaylor reviewed (changes requested):
Can we also delete the references to the remove symbols in tests/configs/crypto_config_test_driver_extension.h, as these presumably no longer do anything? Or are these still required?

mpg commented on drivers/builtin/src/ecp.c:
```diff
@@ -3407,8 +3401,8 @@ int mbedtls_ecp_self_test(int verbose)
     /* Use secp192r1 if available, or any available curve */
-#if defined(MBEDTLS_ECP_DP_SECP192R1_ENABLED)
-    MBEDTLS_MPI_CHK(mbedtls_ecp_group_load(&grp, MBEDTLS_ECP_DP_SECP192R1));
+#if defined(MBEDTLS_ECP_DP_SECP256R1_ENABLED)
```
Note: the comment above should be updated

mpg reviewed (changes requested):
Looking pretty good, but `git grep -Ei '[KR]1_192|SECP192'` still finds a few hits that I think should be removed too, in:
- `configs/ext/crypto_config_profile_medium.h`
- `drivers/builtin/include/mbedtls/private/ecp.h`
- `drivers/builtin/src/crypto_oid.h`
- `drivers/builtin/src/ecp.c`
- `include/psa/crypto_values.h`
- `tests/configs/crypto_config_test_driver_extension.h`

Also, the CI's not fully happy yet.

valeriosetti committed:
core: check_config: fix typo on secp192r1 curve

The check should be for both secp192r1 and secp192k1, whereas before it
was using k1 twice. That was clearly a mistake.

Signed-off-by: Valerio Setti <vsetti@baylibre.com>

valeriosetti committed:
drivers: ecp: fix comment still mentioning secp192r1

It should have been changed to secp256r1 when the code below was updated.
Doing it now.

Signed-off-by: Valerio Setti <vsetti@baylibre.com>

valeriosetti committed:
remove remaining mentions to secp192[k|r]1 from comments

Signed-off-by: Valerio Setti <vsetti@baylibre.com>

valeriosetti committed:
oid: remove support for secp192[r|k]1

Signed-off-by: Valerio Setti <vsetti@baylibre.com>

valeriosetti committed:
config_test_driver_extension: remove support for secp192[k|r]1

Signed-off-by: Valerio Setti <vsetti@baylibre.com>

valeriosetti committed:
drivers: ecp: remove support for secp192[k|r]1 from ecp_group_id

Signed-off-by: Valerio Setti <vsetti@baylibre.com>

valeriosetti commented:
> Looking pretty good, but `git grep -Ei '[KR]1_192|SECP192'` still finds a few hits that I think should be removed too, in:

I apologize for the mistake, I should have removed all occurrences now. Running the command
`rg -ie "[krKR]1_192|(secp|SECP)192"`
only returns occurrences in comments that still make sense.

valeriosetti commented:
> Also, the CI's not fully happy yet.

The problem seems to be in `results analysis` here:
```
17:16:01  *** Analyze coverage ***
17:16:07  Error: Test case not executed: test_suite_x509parse;X509 CRT ASN1 (ECDSA signature, EC key)
17:16:07  Error: Test case not executed: test_suite_x509parse;X509 CRT ASN1 (RSA signature, EC key)
17:16:07  Error: Test case not executed: test_suite_x509parse;X509 CRT information EC signed by RSA
17:16:07  Error: Test case not executed: test_suite_x509parse;X509 CRT verification #32 (Valid, EC cert, RSA CA)
17:16:07  Error: Test case not executed: test_suite_x509parse;X509 CRT verification #93 (Suite B invalid, EC cert, RSA CA)
17:16:07  Error: Test case not executed: test_suite_x509parse;X509 CRT verify chain #12 (suiteb profile, RSA root)
```

which should be resolved now by https://github.com/Mbed-TLS/mbedtls/pull/10522. Hopefully the new CI run will pick this change and be happy.

valeriosetti committed:
tests: pk: add temporary fix for the removal of secp192 curves

This fix will be reverted in #591.

Signed-off-by: Valerio Setti <vsetti@baylibre.com>

valeriosetti committed:
framework: update reference

This is unnecessary for the content of this PR, but is required to make
the CI happy. It just updates the framework pointer to the HEAD of
main branch.

Signed-off-by: Valerio Setti <vsetti@baylibre.com>

mpg reviewed (approved):
LGTM, thanks!
